src/background/background-events.js

```diff
@@ -18,12 +19,24 @@ const listener = async ({ type, detail }, port) => {
       });
     });
   }
-
   return true;
 };
 
 export const listen = () => {
   browser.runtime.onConnect.addListener((port) => {
     port.onMessage.addListener(listener);
   });
+
+  browser.tabs.onUpdated.addListener(async (tabId, changeInfo) => {
+    if (changeInfo.url) {
+      for (const [domainName, pattern] of domains) {
+        if (changeInfo.url.match(pattern)) {
+          browser.tabs.sendMessage(tabId, {
+            type: EVENT_URL_CHANGED,
+            detail: domainName,
+          });
+        }
+      }
+    }
+  });
```

Comment:
I think you don't need the `domain` thing.

The background will send a message every time the URL is changed, and the Content Script will listen to that message. Or not.

Reply:
Alright, I removed this part.